src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,264 @@
+package duke.util;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeInvalidArgumentFormatException;
+import duke.exception.DukeInvalidDateFormatException;
+import duke.exception.DukeUnknownKeywordException;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+/*
+ * Parser
+ *
+ * CS2103 AY19/20 Semester 2
+ * Individual Project
+ * Duke Project
+ *
+ * 28 Jan 2020
+ *
+ */
+
+/**
+ * <p>Parser class describes the behaviour of a
+ * parser that parses commands entered by the client.</p>
+ * @author Mario Lorenzo
+ */
+
+public class Parser {
```

Comment:
Good use of HashMap to reduce the lookup time of keywords. However as the keywords are quite few, there may not be a need to use hashmap for now. But good job nonetheless!